src/main/java/nextstep/subway/application/LineService.java

```diff
@@ -0,0 +1,71 @@
+package nextstep.subway.application;
+
+import nextstep.subway.domain.Line;
+import nextstep.subway.domain.Station;
+import nextstep.subway.dto.LineRequest;
+import nextstep.subway.dto.LineResponse;
+import nextstep.subway.repository.LineRepository;
+import nextstep.subway.repository.StationRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+@Service
+public class LineService {
+
+    private LineRepository lineRepository;
+
+    private StationRepository stationRepository;
+
+    public LineService(LineRepository lineRepository,
+                       StationRepository stationRepository) {
+        this.lineRepository = lineRepository;
+        this.stationRepository = stationRepository;
+    }
+
+    @Transactional
+    public LineResponse saveLine(LineRequest lineRequest) {
+        Line line = lineRepository.save(lineRequest.toLine());
+        addStation(line, lineRequest.getUpStationId());
+        addStation(line, lineRequest.getDownStationId());
+        return LineResponse.of(line);
+    }
+
+    @Transactional(readOnly = true)
+    public LineResponse findById(Long id) {
+        return LineResponse.of(lineRepository.getById(id));
+    }
+
+    @Transactional(readOnly = true)
+    public List<LineResponse> findAllLines() {
+        List<LineResponse> responses = new ArrayList<>();
+        lineRepository.findAll().stream().forEach(
+                line -> responses.add(LineResponse.of(line))
+        );
+        return responses;
```

Comment:
```suggestion
        return lineRepository.findAll()
                .stream()
                .map(LineResponse::of)
                .collect(Collectors.toList());
```
이렇게 작성해볼수도 있을것같아요! 😀

Reply:
stream을 자주 사용해보지 못했는데, 이렇게도 사용할 수 있겠네요!